src/main/java/com/microsoft/azure/spring/data/cosmosdb/repository/query/AbstractDocumentDbQuery.java

```diff
@@ -27,20 +30,24 @@ public Object execute(Object[] parameters) {
         final ResultProcessor processor = method.getResultProcessor().withDynamicProjection(accessor);
         final String collection = ((DocumentDbEntityMetadata) method.getEntityInformation()).getCollectionName();
 
-        final DocumentDbQueryExecution execution = getExecution(accessor);
+        final DocumentDbQueryExecution execution = getExecution(accessor, processor.getReturnedType());
         return execution.execute(query, processor.getReturnedType().getDomainType(), collection);
     }
 
 
-    private DocumentDbQueryExecution getExecution(DocumentDbParameterAccessor accessor) {
+    private DocumentDbQueryExecution getExecution(DocumentDbParameterAccessor accessor, ReturnedType returnedType) {
         if (isDeleteQuery()) {
             return new DocumentDbQueryExecution.DeleteExecution(operations);
         } else if (method.isPageQuery()) {
             return new DocumentDbQueryExecution.PagedExecution(operations, accessor.getPageable());
         } else if (isExistsQuery()) {
             return new DocumentDbQueryExecution.ExistsExecution(operations);
         } else {
-            return new DocumentDbQueryExecution.MultiEntityExecution(operations);
+            if (method.isCollectionQuery()) {
+                return new DocumentDbQueryExecution.MultiEntityExecution(operations);
+            } else {
+                return new DocumentDbQueryExecution.SingleEntityExecution(operations, returnedType);
+            }
         }
```

Comment:
Is there a reason why the if statement on line 46 is nested inside of the else on line 45 rather than just being an else-if like earlier in the method? I don't think it'd change any functionality and it might match the existing formatting more, but ultimately very minor.

Reply:
no idea, fixed